src/Simulation/CSharpGeneration/RewriteStep.fs

```diff
@@ -61,9 +48,28 @@ type Emitter() =
                 if content <> null then CompilationLoader.GeneratedFile(source, dir, ".dll.g.cs", content) |> ignore
 
             if not compilation.EntryPoints.IsEmpty then
-                let callable = context.allCallables.[Seq.exactlyOne compilation.EntryPoints]
-                let content = EntryPoint.generate context callable
-                CompilationLoader.GeneratedFile(callable.SourceFile, dir, ".EntryPoint.g.cs", content) |> ignore
+
+                let entryPointCallables =
+                    compilation.EntryPoints
+                    |> Seq.map (fun ep -> context.allCallables.[ep])
+
+                let entryPointSources =
+                    entryPointCallables
+                    |> Seq.groupBy (fun ep -> ep.Source.CodeFile)
+
+                let content = EntryPoint.generateMainSource context entryPointCallables
+                let outputFolder = Path.GetFullPath(dir.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar) + Path.DirectorySeparatorChar.ToString());
+                let outputUri = Uri(outputFolder);
+                let fileDir = Path.GetDirectoryName(outputUri.LocalPath);
+                let targetFile = Path.GetFullPath(Path.Combine(fileDir, "EntryPoint.g.Main.cs"));
+                if content <> null then
+                    if not (Directory.Exists(fileDir)) then
+                        Directory.CreateDirectory(fileDir) |> ignore;
+                    File.WriteAllText(targetFile, content);
```

Comment:
For consistency, this should probably go through the Q# compiler to write the file. However, `CompilationLoader.GeneratedFile` only works if the generated file corresponds to a Q# source file. Maybe that method should be changed or a new one added that works for this case?

Reply:
I've added a WriteFile static function to the compilation loader in the compiler repo's branch, and use that here. The other method is commented out for now until we can get the proper coordination for merging all of this into the respective main branches.